src/App.tsx

```diff
@@ -1,73 +1,146 @@
-import React from 'react';
+import React, {
+  useCallback,
+  useMemo,
+  useState,
+  useRef,
+  useEffect,
+} from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import classNames from 'classnames';
+import debounce from 'lodash.debounce';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const people: Person[] = peopleFromServer;
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isDropdownActive, setIsDropdownActive] = useState(true);
+  const [chosenPerson, setChosenPerson] = useState<Person | null>(null);
+  const delay = 300;
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, delay), [
+    setAppliedQuery,
+    delay,
+  ]);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+
+    setQuery(value);
+    applyQuery(value);
+    setChosenPerson(null);
+    setIsDropdownActive(true);
+  };
+
+  const handlePersonClick = (slug: string) => {
+    const foundedPerson = people.find(person => person.slug === slug) || null;
+
+    setChosenPerson(foundedPerson);
+    setIsDropdownActive(false);
+  };
+
+  const handleInputClick = () => {
+    setIsDropdownActive(true);
+  };
+
+  const filteredPeople = useMemo(() => {
+    return people.filter(person =>
+      person.name.toLowerCase().includes(appliedQuery.toLowerCase()),
+    );
+  }, [appliedQuery, people]);
+
+  // test
+  const timeoutRef = useRef<number | null>(null);
+
+  const handleInputBlur = () => {
+    timeoutRef.current = window.setTimeout(() => {
+      setIsDropdownActive(false);
+    }, 100);
+  };
+
+  const handleInputFocus = () => {
+    if (timeoutRef.current !== null) {
```

Comment:
```suggestion
    if (timeoutRef.current) {
```

can be a bit simplify